Flake8 3.2.0 broke the build

```
(karellen-venv) bash-4.3$ pip install --upgrade flake8
Collecting flake8
  Downloading flake8-3.2.0-py2.py3-none-any.whl (66kB)
    100% |████████████████████████████████| 71kB 683kB/s 
Collecting pycodestyle<2.3.0,>=2.0.0 (from flake8)
  Downloading pycodestyle-2.2.0-py2.py3-none-any.whl (44kB)
    100% |████████████████████████████████| 51kB 4.2MB/s 
Requirement already up-to-date: mccabe<0.6.0,>=0.5.0 in /home/arcivanov/.pyenv/versions/3.5.2/envs/karellen-venv/lib/python3.5/site-packages (from flake8)
Collecting pyflakes!=1.2.0,!=1.2.1,!=1.2.2,<1.4.0,>=0.8.1 (from flake8)
  Downloading pyflakes-1.3.0-py2.py3-none-any.whl (222kB)
    100% |████████████████████████████████| 225kB 2.6MB/s 
Installing collected packages: pycodestyle, pyflakes, flake8
  Found existing installation: pycodestyle 2.0.0
    Uninstalling pycodestyle-2.0.0:
      Successfully uninstalled pycodestyle-2.0.0
  Found existing installation: pyflakes 1.2.3
    Uninstalling pyflakes-1.2.3:
      Successfully uninstalled pyflakes-1.2.3
  Found existing installation: flake8 3.0.4
    Uninstalling flake8-3.0.4:
      Successfully uninstalled flake8-3.0.4
Successfully installed flake8-3.2.0 pycodestyle-2.2.0 pyflakes-1.3.0
(karellen-venv) bash-4.3$ ./build.py
PyBuilder version ${dist_version}
Build started at 2016-11-17 03:07:01
------------------------------------------------------------
[INFO]  Building pybuilder version 0.11.10.dev (0.11.10.dev20161117080701)
[INFO]  Executing build in /home/arcivanov/Documents/src/arcivanov/pybuilder
[INFO]  Going to execute tasks: install_dependencies, analyze, publish
[INFO]  Installing all dependencies
[INFO]  Processing batch dependency 'pip>=7.1'
[INFO]  Running unit tests
[INFO]  Executing unit tests from Python modules in /home/arcivanov/Documents/src/arcivanov/pybuilder/src/unittest/python
[INFO]  Executed 633 unit tests
[INFO]  All unit tests passed.
[INFO]  Executing flake8 on project sources.
[WARN]  flake8: src/integrationtest/python/should_build_pyfix_project_tests.py:63:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_build_simple_python_project_tests.py:56:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_exclude_optional_task_tests.py:53:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_filter_resources_tests.py:48:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_invoke_initializer_when_environments_match_tests.py:47:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_write_manifest_file_tests.py:70:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_list_multiple_tasks_for_project_using_core_plugin_tests.py:43:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_list_multiple_tasks_for_simple_project_tests.py:47:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_list_single_task_for_simple_project_tests.py:39:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_not_invoke_initializer_when_environment_do_not_match_tests.py:45:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_raise_exception_when_no_default_goal_is_given_tests.py:39:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_raise_exception_when_project_is_not_valid_tests.py:42:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_set_pyfix_glob_from_suffix_tests.py:69:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_skip_non_teardown_on_task_failure_tests.py:51:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_skip_undefined_keys_when_filtering_tests.py:46:1: E305 expected 2 blank lines after class or function definition, found 1
[WARN]  flake8: src/integrationtest/python/should_suppress_error_during_teardown_tests.py:50:1: E305 expected 2 blank lines after class or function definition, found 1
------------------------------------------------------------
BUILD FAILED - flake8 found 16 warning(s)
------------------------------------------------------------
Build finished at 2016-11-17 03:07:07
Build took 5 seconds (5758 ms)
```

--- a/src/integrationtest/python/should_build_pyfix_project_tests.py
+++ b/src/integrationtest/python/should_build_pyfix_project_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin
@@ -59,6 +58,7 @@ def should_skip_test_sans_pyfix_test ():
             "target/reports/pyfix_unittest.json", '"failures": []')
         self.assert_file_contains(
             "target/reports/pyfix_unittest.json", '"tests-run": 1')
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_build_simple_python_project_tests.py
+++ b/src/integrationtest/python/should_build_simple_python_project_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin
@@ -52,6 +51,7 @@ def spam ():
 def spam ():
     pass
 """)
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_exclude_optional_task_tests.py
+++ b/src/integrationtest/python/should_exclude_optional_task_tests.py
@@ -50,5 +50,6 @@ def task_c(project):
         self.assertFalse(project.get_property("a"))
         self.assertTrue(project.get_property("c"))
 
+
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_filter_resources_tests.py
+++ b/src/integrationtest/python/should_filter_resources_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin, init
@@ -44,6 +43,7 @@ def init (project):
 
         self.assert_file_content("target/spam", "1.0.dev0")
         self.assert_file_content("target/eggs", "${version}")
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_invoke_initializer_when_environments_match_tests.py
+++ b/src/integrationtest/python/should_invoke_initializer_when_environments_match_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import init, task
@@ -43,6 +42,7 @@ def any_task (project):
 
         reactor = self.prepare_reactor()
         reactor.build(environments=["test_environment"])
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_list_multiple_tasks_for_project_using_core_plugin_tests.py
+++ b/src/integrationtest/python/should_list_multiple_tasks_for_project_using_core_plugin_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin
@@ -39,6 +38,7 @@ use_plugin("core")
 
         self.assertTrue("clean" in task_names)
         self.assertTrue("publish" in task_names)
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_list_multiple_tasks_for_simple_project_tests.py
+++ b/src/integrationtest/python/should_list_multiple_tasks_for_simple_project_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import task
@@ -43,6 +42,7 @@ def my_task (): pass
 
         self.assertEqual(
             ["a_task_with_overridden_name", "another_task", "my_task"], sorted(actual_task_names))
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_list_single_task_for_simple_project_tests.py
+++ b/src/integrationtest/python/should_list_single_task_for_simple_project_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import task
@@ -35,6 +34,7 @@ def my_task (): pass
         tasks = reactor.get_tasks()
         self.assertEquals(1, len(tasks))
         self.assertEquals("my_task", tasks[0].name)
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_not_invoke_initializer_when_environment_do_not_match_tests.py
+++ b/src/integrationtest/python/should_not_invoke_initializer_when_environment_do_not_match_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import init, task
@@ -41,6 +40,7 @@ def any_task (): pass
 
         reactor = self.prepare_reactor()
         reactor.build()
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_raise_exception_when_no_default_goal_is_given_tests.py
+++ b/src/integrationtest/python/should_raise_exception_when_no_default_goal_is_given_tests.py
@@ -19,12 +19,10 @@
 import unittest
 
 from integrationtest_support import IntegrationTestSupport
-
 from pybuilder.errors import PyBuilderException
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import task
@@ -35,6 +33,7 @@ def spam (): pass
         reactor = self.prepare_reactor()
 
         self.assertRaises(PyBuilderException, reactor.build)
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_raise_exception_when_project_is_not_valid_tests.py
+++ b/src/integrationtest/python/should_raise_exception_when_project_is_not_valid_tests.py
@@ -19,12 +19,10 @@
 import unittest
 
 from integrationtest_support import IntegrationTestSupport
-
 from pybuilder.errors import ProjectValidationFailedException
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import init
@@ -38,6 +36,7 @@ def init (project):
 
         self.assertRaises(
             ProjectValidationFailedException, reactor.build, ["clean"])
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_set_pyfix_glob_from_suffix_tests.py
+++ b/src/integrationtest/python/should_set_pyfix_glob_from_suffix_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import init
@@ -65,6 +64,7 @@ raise Exception("This test should not have run!")
             "target/reports/pyfix_unittest.json", '"failures": []')
         self.assert_file_contains(
             "target/reports/pyfix_unittest.json", '"tests-run": 1')
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_skip_non_teardown_on_task_failure_tests.py
+++ b/src/integrationtest/python/should_skip_non_teardown_on_task_failure_tests.py
@@ -48,5 +48,6 @@ def teardown_foo(project):
         self.assertTrue(project.get_property("non_teardown_foo ran") is None)
         self.assertTrue(project.get_property("teardown_foo completed"))
 
+
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_skip_undefined_keys_when_filtering_tests.py
+++ b/src/integrationtest/python/should_skip_undefined_keys_when_filtering_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin, init
@@ -42,6 +41,7 @@ def init (project):
         reactor.build("package")
 
         self.assert_file_content("target/spam", "1.0.dev0 ${any_undefined_key}")
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_suppress_error_during_teardown_tests.py
+++ b/src/integrationtest/python/should_suppress_error_during_teardown_tests.py
@@ -47,5 +47,6 @@ def teardown2_foo(project):
         self.assertRaises(ValueError, reactor.build, "foo")
         self.assertTrue(project.get_property("teardown2_foo completed"))
 
+
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_write_manifest_file_tests.py
+++ b/src/integrationtest/python/should_write_manifest_file_tests.py
@@ -21,8 +21,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin, init
@@ -66,6 +65,7 @@ def init (project):
 include more_spam
 include more_eggs
 """)
+
 
 if __name__ == "__main__":
     unittest.main()

--- a/src/integrationtest/python/should_write_setup_file_tests.py
+++ b/src/integrationtest/python/should_write_setup_file_tests.py
@@ -19,8 +19,7 @@ import unittest
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         self.write_build_file("""
 from pybuilder.core import use_plugin, init

--- a/src/integrationtest/python/should_write_setup_file_with_requirements_tests.py
+++ b/src/integrationtest/python/should_write_setup_file_with_requirements_tests.py
@@ -20,8 +20,7 @@ from os.path import join
 from integrationtest_support import IntegrationTestSupport
 
 
-class Test (IntegrationTestSupport):
-
+class Test(IntegrationTestSupport):
     def test(self):
         requirements = join(self.tmp_directory, "requirements.txt")
         self.write_build_file("""

--- a/src/main/python/pybuilder/plugins/python/flake8_plugin.py
+++ b/src/main/python/pybuilder/plugins/python/flake8_plugin.py
@@ -36,7 +36,7 @@ use_plugin("python.core")
 
 @init
 def initialize_flake8_plugin(project):
-    project.plugin_depends_on("flake8")
+    project.plugin_depends_on("flake8", "~=3.2")
 
     project.set_property_if_unset("flake8_break_build", False)
     project.set_property_if_unset("flake8_max_line_length", 120)

--- a/src/unittest/python/plugins/python/flake8_plugin_tests.py
+++ b/src/unittest/python/plugins/python/flake8_plugin_tests.py
@@ -12,7 +12,7 @@ class FlakePluginInitializationTests(TestCase):
     def test_should_set_dependency(self):
         mock_project = Mock(Project)
         initialize_flake8_plugin(mock_project)
-        mock_project.plugin_depends_on.assert_called_with('flake8')
+        mock_project.plugin_depends_on.assert_called_with('flake8', "~=3.2")
 
     def test_should_leave_user_specified_properties_when_initializing_plugin(self):
 

